In [14]:
import pandas as pd 
import numpy as np
from gensim.models.word2vec import Word2Vec
import nltk
from nltk.tokenize import word_tokenize

In [7]:
bbc_data = pd.read_csv(r"C:\Users\Home\Downloads\bbc-text.csv")

In [20]:
bbc_data

category                                               text
0              tech  tv future in the hands of viewers with home th...
1          business  worldcom boss  left books alone  former worldc...
2             sport  tigers wary of farrell  gamble  leicester say ...
3             sport  yeading face newcastle in fa cup premiership s...
4     entertainment  ocean s twelve raids box office ocean s twelve...
...             ...                                                ...
2220       business  cars pull down us retail figures us retail sal...
2221       politics  kilroy unveils immigration policy ex-chatshow ...
2222  entertainment  rem announce new glasgow concert us band rem h...
2223       politics  how political squabbles snowball it s become c...
2224          sport  souness delight at euro progress boss graeme s...

[2225 rows x 2 columns]

There are 2225 datapoints and 2 cloumns 

 There are 5 unique category ('tech', 'business', 'sport', 'entertainment', 'politics'), for each category there are many text given 

In [8]:
list_of_sent = []
for i in bbc_data['text']:
    list_of_sent.append(i.split())

i have split each and every word from the column text in bbc dataframe 

In [9]:
w2v_model = Word2Vec(list_of_sent,vector_size=10,min_count=5)

here i am training the model with all words taken from text column in bbc_data dataframe as my input data size is less i have decreased the vectoer size of the words to 10 , if the word is occured minimun 5 times in total list_of_sent list then that word will be considered and trained to the model 

In [10]:
words = list(w2v_model.wv.key_to_index)

this command will give all the words for which the model is trained and given vectors , i have stored all those words into a variable 'words'

In [17]:
sent_vecs = []
for i in bbc_data['text'][0:250]:
    sent_array = []
    count = 0
    for j in word_tokenize(i):
        if j in words:
            sent_array.append(w2v_model.wv[j])
            count +=1
    sent_vecs.append(np.sum(sent_array,axis=0)/count)

in word2vec only the words are converted to vectors but we have convert whole sentance to vector , so we use one of the technique avgword2vec so that we can convert whole sentence in to array of vector

i took first 250 rows from the bbc_data and converted whole text column in those rows into vector form , here evey word dimension is 10 as we set in the above command so the dimension of sentance will be also 10 as we add up all the words in the sentance to get that sentance vector 

In [24]:
sent_vecs

[array([ 0.8490572 , -0.67629695,  0.08830237,  0.7550186 ,  0.86115533,
        -0.24625021,  1.4660021 ,  0.90549994, -1.6063116 ,  0.29067802],
       dtype=float32),
 array([ 0.29254568, -0.07582706,  0.80345994, -0.10308409,  0.7746775 ,
         0.2630343 ,  1.6659956 ,  1.2089123 , -0.87826717,  0.3687065 ],
       dtype=float32),
 array([ 0.92295766, -0.00174685,  0.1760557 ,  0.24129617,  0.8953543 ,
         0.3049884 ,  1.3507292 ,  1.3776441 , -1.4930316 ,  0.44041222],
       dtype=float32),
 array([ 0.39418575, -0.20794657,  0.12879756, -0.24716677,  0.42772576,
         0.81361073,  1.4727069 ,  0.8270455 , -1.3074347 , -0.9245858 ],
       dtype=float32),
 array([-0.16811171, -0.03846965,  0.22416775,  0.15123242,  0.40032935,
         0.48209375,  1.6701262 ,  0.70902395, -1.2491629 , -0.36058065],
       dtype=float32),
 array([ 0.78605175,  0.16630127,  0.68293023,  0.15692529,  1.1470246 ,
         0.3787984 ,  1.701128  ,  1.6072451 , -0.90238804,  0.7500165 ],
   